lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp

```diff
@@ -1878,6 +1878,26 @@ static LogicalResult printOperation(CppEmitter &emitter,
   return success();
 }
 
+// Print an expand shape by forwarding the value to the next op
+static LogicalResult printOperation(CppEmitter &emitter,
+                                    memref::ExpandShapeOp expandShapeOp) {
+  Value source = expandShapeOp.getSrc();
+
+  // If the memref being outputted is not already emitted,
+  // error out
+  if (!emitter.hasValueInScope(source))
+    return failure();
+
+  if (failed(emitter.emitAssignPrefix(*expandShapeOp)))
+    return failure();
+
+  raw_indented_ostream &os = emitter.ostream();
+
+  os << emitter.getOrCreateName(source);
```

Comment:
For my own understanding. This op essentially is a no-op in the cpp? It only does variable assignment like "v18 = v17"?

Reply:
Indeed, it's just forwarding the value, and [that is the semantics of the operation itself](https://mlir.llvm.org/docs/Dialects/MemRef/#memrefexpand_shape-memrefexpandshapeop): _"The operation is limited to such reassociations, where a dimension is expanded into one or multiple contiguous dimensions. Such reassociations never require additional allocs or copies."_

A better option would be to replace all uses of the op with the source value and call it a day, so you don't give extra work to the C++ compiler, but this was a more immediate solution. I still might give it a go before landing it.

Reply:
I see, that sounds reasonable to me. Hopefully, our C++ compiler can figure that out. If not, we can go for your better option later.